File: 11-container-with-most-water/Container_With_Most_Water.cpp
class Solution {
public:
    int maxArea(vector<int>& h) {
        int maxi = 0;
        int i=0,j=h.size()-1;
        while(i<j){
            maxi = max((min(h[i],h[j])*(j-i)),maxi);
            if(h[i]<h[j]){
                i++;
            }
            else{
                j--;
            }
        }
        return maxi;
    }
};

File: Container With Most Water.java
public class Solution {
    public int maxArea(int[] A) {
        if(A.length<=1)return 0;
        int ans=Integer.MIN_VALUE;
        int i=0;
        int j=A.length-1;
        while(i<j){
            ans=Math.max(ans,(j-i)*Math.min(A[i],A[j]));
            if(A[i]<A[j]){
                i++;
            }else{
                j--;
            }
        }
        return ans;
    }
}
